internal/pkg/gateway/commit/notifier_test.go

```diff
@@ -0,0 +1,299 @@
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	deliverMock "github.com/hyperledger/fabric/common/deliver/mock"
+	"github.com/hyperledger/fabric/internal/pkg/gateway/commit"
+	commitMocks "github.com/hyperledger/fabric/internal/pkg/gateway/commit/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNotifier(t *testing.T) {
+	type Transaction struct {
+		ID     string
+		Status peerProto.TxValidationCode
+	}
+
+	NewPayload := func(transactionID string) *commonProto.Payload {
+		channelHeader := &commonProto.ChannelHeader{
+			Type: int32(commonProto.HeaderType_ENDORSER_TRANSACTION),
+			TxId: transactionID,
+		}
+		channelHeaderBytes, err := proto.Marshal(channelHeader)
+		assert.NoError(t, err)
+
+		return &commonProto.Payload{
+			Header: &commonProto.Header{
+				ChannelHeader: channelHeaderBytes,
+			},
+		}
+	}
+
+	NewTransactionEnvelope := func(transaction *Transaction) *commonProto.Envelope {
+		payload := NewPayload(transaction.ID)
+		payloadBytes, err := proto.Marshal(payload)
+		assert.NoError(t, err)
+
+		return &commonProto.Envelope{
+			Payload: payloadBytes,
+		}
+	}
+
+	NewBlock := func(t *testing.T, transactions ...*Transaction) *commonProto.Block {
+		var validationCodes []byte
+		var envelopes [][]byte
+
+		for _, transaction := range transactions {
+			validationCodes = append(validationCodes, byte(transaction.Status))
+
+			envelope := NewTransactionEnvelope(transaction)
+			envelopeBytes, err := proto.Marshal(envelope)
+			assert.NoError(t, err)
+			envelopes = append(envelopes, envelopeBytes)
+		}
+
+		metadata := make([][]byte, len(commonProto.BlockMetadataIndex_name))
+		metadata[int(commonProto.BlockMetadataIndex_TRANSACTIONS_FILTER)] = validationCodes
+
+		return &commonProto.Block{
+			Header: &commonProto.BlockHeader{
+				Number: 1,
+			},
+			Metadata: &commonProto.BlockMetadata{
+				Metadata: metadata,
+			},
+			Data: &commonProto.BlockData{
+				Data: envelopes,
+			},
+		}
+	}
+
+	NewMockBlockReader := func(blocks ...*commonProto.Block) *commitMocks.BlockReader {
+		blockReader := &commitMocks.BlockReader{}
+
+		iterator := &deliverMock.BlockIterator{}
+		blockReader.IteratorReturns(iterator, nil)
+
+		for i, block := range blocks {
+			iterator.NextReturnsOnCall(i, block, commonProto.Status_SUCCESS)
+		}
+
+		return blockReader
+	}
+
+	NewTestNotifier := func(blocks ...*commonProto.Block) *commit.Notifier {
+		blockReader := NewMockBlockReader(blocks...)
+		return commit.NewNotifier(blockReader)
+	}
+
+	t.Run("Notify returns error if block iterator cannot be obtained", func(t *testing.T) {
+		expected := errors.New("MY_ERROR_MESSAGE")
+		blockReader := &commitMocks.BlockReader{}
+		blockReader.IteratorReturns(nil, expected)
+		notifier := commit.NewNotifier(blockReader)
+
+		_, err := notifier.Notify("channel", "transactionID")
+		assert.ErrorContains(t, err, expected.Error())
```

Comment:
For tests, we generally prefer assertions to be made over literals where possible, rather than deriving them.  (It also gets rid of the need to store the intermediate error, so saves you a line in your setup).

Reply:
👍